### PR TITLE
[O11y][Salesforce] Add support of the `interval` parameter in Salesforce `setupaudittrail-rest` fileset.

### LINKED ISSUE #35917: [Filebeat][Salesforce] Add support of interval parameter in Salesforce setupaudittrail-rest fileset
**Describe the enhancement:**
- add support of interval parameter in Salesforce setupaudittrail-rest fileset.
- Improve documentation for the same to explain more in detail how salesforce data collection frequency works.

**Describe a specific use case for the enhancement or feature:**
- User would be able to configure how frequently they want to collect setup-audit-trail events from salesforce.
- User would be able to to configure it based on the data pool size from salesforce based on need.

Related:
- https://github.com/elastic/sdh-beats/issues/3531

### PR DESCRIPTION
**Pull request by kush-elastic**
### Urgency
- High

### Activity Type
- Enhancement

## What does this PR do?

- Add support of the `interval` parameter in Salesforce setupaudittrail-rest fileset.
- Improve documentation for the same to explain more in detail how salesforce data collection frequency works for  setupaudittrail-rest fileset.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #35917

### CODE REVIEW

File: filebeat/docs/modules/salesforce.asciidoc [hunk 1]
```diff
@@ -253,6 +254,12 @@ The password used as part of the authentication flow. It is required for authent
 
 The URL of the Saleforce instance.
 
+*`var.interval`*::
+
+Period of fetching logs, i.e. 1s/1m/1h.
```
Comment (lalit-satapathy) [inline filebeat/docs/modules/salesforce.asciidoc:259]: What will be the impact if user selects a lower limit like 1m, as given above?

Comment (kush-elastic) [reply filebeat/docs/modules/salesforce.asciidoc:259]: > What will be the impact if user selects a lower limit like 1m, as given above?

If the user provides a lower limit like 1m, there won't be any issue unless and until the Salesforce REST API rate limit exceeds. I have mentioned the `Note` in which users can find the Salesforce REST API rate limit from the official doc.

**Review by agithomas — APPROVED** [review]
LGTM!

**Comment by lalit-satapathy** [comment]
@kush-elastic,

- Is there any error check in beats code, if user provide a bad value/syntax for the _var.interval_ or salesforce will give error and  the module with exit gracefully? Can you provide the exact errors for this scenario.
- the default value of interval is 1h? Can we make that explicit in the document?
- Still concerned, what happens if user gives low value or very high values? Could not find any specific recommendation in this [doc](https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm) for var.limit

**Comment by kush-elastic** [comment]
> @kush-elastic,
> 
> * Is there any error check in beats code, if user provide a bad value/syntax for the _var.interval_ or salesforce will give error and  the module with exit gracefully? Can you provide the exact errors for this scenario.

Yes, there is an error check in inbuilt beats code. If a user provides a bad value/syntax error for the var.interval. User can see the error.message in the logs.
```
Error message: Exiting: Failed to start crawler: creating module reloader failed: error checking input configuration: runner factory could not check config: time: invalid duration "1hrs" accessing 'interval'
```

> * the default value of interval is 1h? Can we make that explicit in the document?
> * Still concerned, what happens if user gives low value or very high values? Could not find any specific recommendation in this [doc](https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm) for var.limit

We have provided the following `Note` in the document itself. Let me know if the following Note will resolve your concerns!
```
Note: The default value of `var.interval` is 1h. It is important to exercise caution when reducing the interval, as it directly affects the API rate limit of the Salesforce instance. Salesforce API rate limit is ~1000 API calls per hour. Hence if user goes with lower limit of var.interval, the Salesforce API rate limit will exceed and any additional API requests beyond the limit will result in an error response from the Salesforce API. The error message will typically indicate that the rate limit has been exceeded.. Please refer to the following link for the https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm [Salesforce API Rate Limit].

**Comment by lalit-satapathy** [comment]
>Note: The default value of `var.interval` is 1h. It is important to exercise caution when reducing the interval, as it directly affects the API rate limit of the Salesforce instance. Salesforce API rate limit is ~1000 API calls per hour. Hence if user goes with lower limit of var.interval, the Salesforce API rate limit will exceed and any additional API requests beyond the limit will result in an error response from the Salesforce API. The error message will typically indicate that the rate limit has been exceeded.. Please refer to the following link for the https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm [Salesforce API Rate Limit].

Yes this is better.

**Review by milan-elastic — APPROVED** [review]

**Review by lalit-satapathy — APPROVED** [review]
LGTM

**Comment by lalit-satapathy** [comment]
@kush-elastic, Lets raise a seprate PR to update the CODEOWNER for the beats module: /x-pack/filebeat/module/salesforce  to @elastic/obs-infraobs-integrations

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -378,6 +378,7 @@ automatic splitting at root level, if root level element is an array. {pull}3415
 - Improve CEL input documentation {pull}34831[34831]
 - Add metrics documentation for CEL and AWS CloudWatch inputs. {issue}34887[34887] {pull}34889[34889]
 - Metrics hosted by the HTTP monitoring endpoint for the `aws-cloudwatch`, `aws-s3`, `cel`, and `lumberjack` inputs are now available under `/inputs/` instead of `/dataset`.
+- Add support of the interval parameter in Salesforce setupaudittrail-rest fileset. {issue}35917[35917] {pull}35938[35938]
```
Comment (pierrehilbert) [inline CHANGELOG.asciidoc:381]: You shouldn't add an entry here, only in the `CHANGELOG.next.asciidoc`

**Review by pierrehilbert — COMMENTED** [review]
Please remove the `CHANGELOG.asciidoc` entry and I will approve the PR

File: CHANGELOG.asciidoc (hunk 2 above)
Comment (kush-elastic) [reply CHANGELOG.asciidoc:381]: Okay, let me update it.

**Review by pierrehilbert — APPROVED** [review]
LGTM

**Pull request merged**